### PR TITLE
Use `std::vector` and `std::deque` instead of most `std::list`s

### LINKED ISSUE #6779: Consistent naming for `std::list` variables
Current `std::list` variables in our code:

```cpp
std::list<std::shared_ptr<CDemoEdit>> m_lpEditJobs;
std::list<CCacheChunk> m_Cache;
std::list<int> Indices = ...;
std::list<CCharacter *> listOfChars;
std::list<int> m_IDs;
std::list<int> Indices = ...;
std::list<int> Indices;
std::list<std::shared_ptr<CDataFileWriterFinishJob>> m_lpWriterFinishJobs;
std::list<int> Indices = ...;
std::list<CCharacter *> HitCharacters = ...;
std::list<CCharacter *> listOfChars;
```

Option A (prefix for list and its element type):

```
std::list<std::shared_ptr<CDemoEdit>> m_lpEditJobs;
```

Option B (no prefix for list and its element type):

```
std::list<std::shared_ptr<CDemoEdit>> m_EditJobs;
```

Option C (no list prefix, but prefix for the element type):

```
std::list<std::shared_ptr<CDemoEdit>> m_pEditJobs;
```

Option D (replace with `std::vector`)
```
std::vector<std::shared_ptr<CDemoEdit>> m_vpEditJobs;
```

I think we didn't want to go overboard with different prefixes. I definitely don't like option C. I'd slightly prefer A, but B would also be fine. If we don't need any of the special properties of `std::list` then using only `std::vector` would be preferable as well.

### PR DESCRIPTION
**Pull request by Robyt3**
Use `std::vector` in cases where elements are only inserted at the end of the collection.

Use `std::deque` in cases where elements are only inserted/deleted at the beginning/end of the collection.

Use `std::list` in the remaining single case where elements are being removed from arbitrary positions and added at either the beginning or the end of the collection.

Adjust variables names. Don't use separate prefix for `std::deque`s and `std::list`s, as they are only used very rarely. Closes #6779.

## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/engine/client/client.cpp [hunk 1]
```diff
@@ -2879,18 +2879,18 @@ void CClient::Update()
...
+		if(!m_EditJobs.empty())
 		{
-			std::shared_ptr<CDemoEdit> e = m_lpEditJobs.front();
-			if(e->Status() == IJob::STATE_DONE)
+			std::shared_ptr<CDemoEdit> pJob = m_EditJobs.front();
+			if(pJob->Status() == IJob::STATE_DONE)
 			{
-				char aBuf[256];
-				str_format(aBuf, sizeof(aBuf), "Successfully saved the replay to %s!", e->Destination());
+				char aBuf[IO_MAX_PATH_LENGTH + 64];
```
Comment (edg-l) [inline src/engine/client/client.cpp:2887]: why 64?

Comment (def-) [reply src/engine/client/client.cpp:2887]: So you have enough space for some extra text, seems fair enough to me

**Review by def- — APPROVED** [review]
bors r+

File: src/engine/client/client.cpp (hunk 1 above)
Comment (Robyt3) [reply src/engine/client/client.cpp:2887]: It's a nice round number and 32 would have been too small (although not realistically of course).

**Pull request merged**